internal/kafka/internal/services/quota.go

```diff
@@ -11,7 +12,7 @@ type QuotaService interface {
 	// CheckIfQuotaIsDefinedForInstanceType checks if quota is defined for the given instance type
 	CheckIfQuotaIsDefinedForInstanceType(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (bool, *errors.ServiceError)
 	// ReserveQuota reserves a quota for a user and return the reservation id or an error in case of failure
-	ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (string, *errors.ServiceError)
+	ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType, sizeRequired int, kafkaConfig *config.KafkaConfig) (string, *errors.ServiceError)
```

Comment:
The `kafkaConfig *config.KafkaConfig` parameter can be passed on during service instantiation time as a constructor argument. There is no need to pass it here.

Reply:
that was my first thought, but this requires a lot of refactoring of the quota related stuff

Reply:
> but this requires a lot of refactoring of the quota related

Hm the only thing I see happening is passing the default config here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5f404864d106496c3c405cad4735b59b7da0640e/internal/kafka/internal/services/quota/default_quota_service_factory.go#L17 as a constructor argument and use it when instantiating the different quota services in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5f404864d106496c3c405cad4735b59b7da0640e/internal/kafka/internal/services/quota/default_quota_service_factory.go#L23-L24

of course the config will have to be declared in the different structs but it is something you are already doing at: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5f404864d106496c3c405cad4735b59b7da0640e/internal/kafka/internal/services/quota/quota_management_list_service.go#L18. And you'll need to do the same for https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5f404864d106496c3c405cad4735b59b7da0640e/internal/kafka/internal/services/quota/ams_quota_service.go#L14

Reply:
Thank @pawelpaszki